### DISCUSSION TITLE
Recommended Modbus Polling Interval for Inverter Data Collection?

### DISCUSSION THREAD

**Discussion by drumtrich** [WillCodeForCats/solaredge-modbus-multi#697] [General]
Hi everyone,

I'm currently polling Modbus data from my inverter with an interval of 60 seconds. By default, it was set to 300 seconds, which I found a bit high, so I lowered it to capture data more frequently. I'm curious to know what polling intervals others are using and if anyone has experience with higher polling frequencies (e.g., 1 second or 10 seconds). Have you encountered any issues with increased load on the network or performance limitations with high-frequency polling?

Any insights or recommendations would be greatly appreciated. Thank you!

**Comment by Ashpork** [comment]
I'm polling at 5 seconds. Had some issues with the network initially due to running a wireless connected mesh, but now with all the AP points now wired it's a lot more stable.

I do however get the occasional command to the inverter dropped. For example, if I adjust the charge rate, or command mode, I'll get a response "homeassistant.exceptions.HomeAssistantError: Connection to inverter ID 1 failed.". This appears to disappear if I decrease the polling rate, so have just accepted this as a by product of polling frequently and have this scenario handled in my automation/scripts.

**Comment by drumtrich** [comment]
Thank you for your response.
I’ll try setting my frequency to 10 seconds. For automations, it’s definitely more useful to query the values more frequently. All my systems are connected via cable, so I don’t expect any connection issues.

**Comment by Ashpork** [comment]
No problem, I didn't either but still find the occasional drop out at higher polling rates when sending commands to the inverter with a wired connection.

For resilience, and to handle any failed connections, I've wrapped all my updates to the SE fields in retry commands.

https://github.com/amitfin/retry

**Comment by drumtrich** [comment]
That sounds good. Thanks. I'll take a look at the integration.
